packages/zoltan2/test/core/color/coloring1.cpp

```diff
@@ -105,6 +107,112 @@ int validateColoring(RCP<SparseMatrix> A, int *color)
+  int nconflicts = 0;
+  RCP<const SparseMatrix::map_type> rowMap = A->getRowMap();
+  RCP<const SparseMatrix::map_type> colMap = A->getColMap();
+  Vector R = Vector(rowMap);
+  //put colors in the scalar entries of R
+  for(size_t i = 0; i < A->getNodeNumRows(); i++){
+    R.replaceLocalValue(i, color[i]);
+  }
+
+  Vector C = Vector(colMap);
+  Import imp = Import(rowMap, colMap);
+  C.doImport(R, imp, Tpetra::REPLACE);
+
+  Teuchos::ArrayView<const zlno_t> indices;
+  Teuchos::ArrayView<const zscalar_t> values; // not used
+
+  //count conflicts in the graph
+  //loop over local rows, treat local column indices as edges
+  size_t n = A->getNodeNumRows();
+  auto colorData = C.getData();
+  for(size_t i = 0; i < n; i++){
+    A->getLocalRowView(i, indices, values);
+    for(Teuchos_Ordinal j = 0; j < indices.size(); j++){
+      if( ((size_t)indices[j] != i) && (color[i] == colorData[indices[j]]) ){
+        nconflicts++;
+      }
+    }
+  }
+
+  return nconflicts;
+}
+
+int validateDistributedDistance2Coloring(RCP<SparseMatrix> A, int* color){
+  int nconflicts = 0;
+
+  RCP<SparseMatrix> S = rcp(new SparseMatrix(A->getRowMap(), 0));
+  //std::cout<<"S.isFillComplete() = "<<S->isFillComplete()<<"\n";
+  Tpetra::MatrixMatrix::Multiply(*A, true, *A, false, *S);
+  
+  RCP<const SparseMatrix::map_type> rowMap = S->getRowMap();
+  //std::cout<<"rowMap contains\n\t";
+  auto rowIndices = rowMap->getMyGlobalIndices();
+  /*for(int i = 0; i < rowMap->getNodeNumElements(); i++){
+    std::cout<<rowIndices[i]<<" ";
+  }
+  std::cout<<"\n";*/
+  RCP<const SparseMatrix::map_type> colMap = S->getColMap();
+  //std::cout<<"colMap contains\n\t";
+  auto colIndices = colMap->getMyGlobalIndices();
+  /*for(int i = 0; i < colMap->getNodeNumElements(); i++){
+    std::cout<<colIndices[i]<<" ";
+  }
+  std::cout<<"\n";*
+  for(int i = 0; i < colMap->getNodeNumElements(); i++){
+    std::cout<<colMap->getLocalElement(colIndices[i])<<" ";
+  }
+  std::cout<<"\n";*/
+  Vector R = Vector(rowMap);
+  //put the colors in the scalar entries of R.
+  for(size_t i = 0; i < S->getNodeNumRows(); i++){
+    R.replaceLocalValue(i, color[i]);
+  }
+
+  Vector C = Vector(colMap);
+  Import imp = Import(rowMap, colMap);
+  C.doImport(R, imp, Tpetra::REPLACE);
+
+  Teuchos::ArrayView<const zlno_t> indices;
+  Teuchos::ArrayView<const zscalar_t> values; //not used
+
+  //count conflicts in the graph
+  //loop over local rows, treat local column indices as edges
+  size_t n = S->getNodeNumRows();
+  auto colorData = C.getData();
+  for(size_t i = 0; i < n; i++){
+    S->getLocalRowView(i, indices, values);
+    for(Teuchos_Ordinal j = 0; j < indices.size(); j++){
+      if(values[j] == 0) continue;
+      //std::cout<<"vertex "<<indices[j]<<" is color "<<colorData[indices[j]]<<"\n";
+      int rowColor = color[i];
+      int adjColor = 0;
+
+      zgno_t gidx;
+      if(rowMap->getGlobalElement(indices[j]) != Teuchos::OrdinalTraits<zgno_t>::invalid()){
```

Comment:
I don't understand what is going on here.  indices[j] is in the column map of S; why do we check the row map?